build-system/tasks/extension-helpers.js

```diff
@@ -664,14 +652,30 @@ async function buildNpmBinaries(extDir, name, options) {
         wrapper: '',
       },
       bento: {
-        entryPoint: bentoEntryPoint,
+        entryPoint: await getBentoBuildFilename(
+          extDir,
+          getBentoName(name),
+          'web-component',
+          options
+        ),
         outfile: 'web-component.js',
         wrapper: '',
-
-        remap,
-        external,
       },
     };
+
+    // for each bento mode, remap all shared modules and declare them as external
+    // remaps "core" modules to @bentoproject/core
+    // rempas any cross-extension (e.g imports to bento-foo) imports to @bentoproject/foo
+    for (const mode in npm) {
+      const fullEntryPoint = path.join(extDir, npm[mode].entryPoint);
+      const bentoRemaps = getRemapBentoNpmDependencies(fullEntryPoint);
+      const bentoExternals = Object.values(bentoRemaps);
+      npm[mode].remap = {
+        ...(npm[mode].remap || {}),
+        ...bentoRemaps,
+      };
+      npm[mode].external = [...(npm[mode].external || []), ...bentoExternals];
+    }
```

Comment:
Do you think it's easier to read if `remap` and `external` is set as each mode of npm is declared? 
```
npm = {
      preact: {
        entryPoint: 'component.js',
        outfile: 'component-preact.js',
        external: ['preact', 'preact/dom', 'preact/compat', 'preact/hooks', ...bentoExternals],
        remap: {'preact/dom': 'preact', ...bentoRemaps},
        wrapper: '',
      }
}
```

This way you can avoid some repetition too i.e. calling `getRemapBentoNpmDependencies` for every mode.

Reply:
Oops, I missed that entryPoint is different for each.

Reply:
yeahh had to move it out because they were all different and it was cleaner that way